src/com/puppetlabs/mq.clj

```diff
@@ -132,10 +132,16 @@
     (start-broker! (build-embedded-broker brokername dir config))
     (catch java.io.EOFException e
       (log/warn
-        (str "Caught EOFException on broker startup, trying to restart it "
-             "again to see if that solves it. This is probably due to "
-             "KahaDB corruption."))
-      (start-broker! (build-embedded-broker brokername dir config)))))
+       "Caught EOFException on broker startup, trying again."
+       "This is probably due to KahaDB corruption "
+       "(see \"KahaDB Corruption\" in the PuppetDB manual).")
+      (start-broker! (build-embedded-broker brokername dir config)))
+    (catch java.io.IOException e
+      (log/error
+       e "Unable to start broker in " (pr-str dir) "."
+       "This is probably due to KahaDB corruption or version downgrade "
```

Comment:
I meant in addition to the edits in the kahadb doc in this PR. I don't mind including more in the error, just seems like "corruption" captures both cases.